src/phpDebug.ts

```diff
@@ -240,11 +242,19 @@ class PhpDebugSession extends vscode.DebugSession {
                     connection.on('close', disposeConnection);
                     const initPacket = await connection.waitForInitPacket();
                     this.sendEvent(new vscode.ThreadEvent('started', connection.id));
-                    await connection.sendFeatureSetCommand('max_depth', '5');
-                    // raise default of 32
-                    await connection.sendFeatureSetCommand('max_children', '100');
-                    // don't truncate long variable values
-                    await connection.sendFeatureSetCommand('max_data', semver.lt(initPacket.engineVersion.replace(/((?:dev|alpha|beta|RC|stable)\d*)$/, '-$1'), '2.2.4') ? '10000' : '0');
+
+                    // setup Xdebug from launch configuration
+                    const xdebugSettings = Object.assign(
+                        {
+                            // don't truncate long variable values
+                            max_data: semver.lt(initPacket.engineVersion.replace(/((?:dev|alpha|beta|RC|stable)\d*)$/, '-$1'), '2.2.4') ? '10000' : '0'
+                        },
+                        args.xdebugSettings || {}
+                    );
+                    for (let xdebugCommand of Object.keys(xdebugSettings)) {
+                        await connection.sendFeatureSetCommand(xdebugCommand, xdebugSettings[xdebugCommand]);
+                    }
+
                     // request breakpoints from VS Code
                     await this.sendEvent(new vscode.InitializedEvent());
                 } catch (error) {
```

Comment:
Because this is the handler for a single connection, and we can have multiple connections. Just because on connection errors doesn't mean we should shutdown the whole debug adapter